### PR TITLE
lib: add the tmpDir parameter

### PR DESCRIPTION
**Pull request by alban**
Fixes: https://github.com/appc/docker2aci/issues/50

### CODE REVIEW

File: lib/docker2aci.go [hunk 1]
```diff
@@ -103,7 +107,7 @@ func convertReal(backend Docker2ACIBackend, dockerURL string, squash bool, outpu
 	}
 
 	layersOutputDir := outputDir
-	if squash {
+	if squash && tmpDir == "" {
```
Comment (jonboulle) [inline lib/docker2aci.go]: hmm? Shouldn't you be passing this into ioutil.TempDir below?

Comment (alban) [inline lib/docker2aci.go]: You're right of course.

File: lib/docker2aci.go [hunk 2]
```diff
@@ -53,11 +53,13 @@ type Docker2ACIBackend interface {
 // If the squash flag is true, it squashes all the layers in one file and
 // places this file in outputDir; if it is false, it places every layer in its
 // own ACI in outputDir.
+// It will use the temporary directory specified by tmpDir, or the default
+// temporary directory if tmpDir is "".
 // username and password can be passed if the image needs authentication.
 // It returns the list of generated ACI paths.
-func Convert(dockerURL string, squash bool, outputDir string, username string, password string) ([]string, error) {
+func Convert(dockerURL string, squash bool, outputDir string, tmpDir string, username string, password string) ([]string, error) {
```
Comment (krnowak) [inline lib/docker2aci.go]: Function with six parameters, ugh. This is getting out of hand.

Feel free to ignore this comment, of course.

Comment (jonboulle) [inline lib/docker2aci.go]: I agree :/

**Comment by jonboulle** [comment]
LGTM unless you would like to clean up the args.

**Comment by alban** [comment]
Patch rebased on master and merged.

**Pull request merged**